src/trackinfoobject.cpp

```diff
@@ -858,64 +815,69 @@ void TrackInfoObject::slotCueUpdated() {
     emit(cuesUpdated());
 }
 
-Cue* TrackInfoObject::addCue() {
+CuePointer TrackInfoObject::addCue() {
     //qDebug() << "TrackInfoObject::addCue()";
     QMutexLocker lock(&m_qMutex);
-    Cue* cue = new Cue(m_iId);
-    connect(cue, SIGNAL(updated()),
+    CuePointer pCue(new Cue(m_iId));
+    connect(pCue.data(), SIGNAL(updated()),
             this, SLOT(slotCueUpdated()));
-    m_cuePoints.push_back(cue);
-    setDirty(true);
-    lock.unlock();
+    m_cuePoints.push_back(pCue);
+    setDirtyAndUnlock(&lock, true);
     emit(cuesUpdated());
-    return cue;
+    return pCue;
 }
 
-void TrackInfoObject::removeCue(Cue* cue) {
+void TrackInfoObject::removeCue(CuePointer pCue) {
     QMutexLocker lock(&m_qMutex);
-    disconnect(cue, 0, this, 0);
-    // TODO(XXX): Delete the cue point.
-    m_cuePoints.removeOne(cue);
-    setDirty(true);
-    lock.unlock();
+    disconnect(pCue.data(), 0, this, 0);
+    m_cuePoints.removeOne(pCue);
+    setDirtyAndUnlock(&lock, true);
     emit(cuesUpdated());
 }
 
-const QList<Cue*>& TrackInfoObject::getCuePoints() {
+QList<CuePointer> TrackInfoObject::getCuePoints() const {
     QMutexLocker lock(&m_qMutex);
     return m_cuePoints;
 }
 
-void TrackInfoObject::setCuePoints(QList<Cue*> cuePoints) {
+void TrackInfoObject::setCuePoints(QList<CuePointer> /*by-value*/ cuePoints) {
     //qDebug() << "setCuePoints" << cuePoints.length();
     QMutexLocker lock(&m_qMutex);
-    QListIterator<Cue*> it(m_cuePoints);
-    while (it.hasNext()) {
-        Cue* cue = it.next();
-        disconnect(cue, 0, this, 0);
+    {
```

Comment:
Maybe I planned to move those parts into separate functions. And the unused iterator is destroyed after the loop has finished. I will remove the extra scope if this is confusing.